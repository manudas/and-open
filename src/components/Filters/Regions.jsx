import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';

import {
    withStyles,
} from '@material-ui/core/styles';

import * as options from '../../data/userDataOptions';

const StyledTypography = withStyles({
    root: {
        marginBottom: '1.3em',
        color: '#9a9a9a',
        fontSize: '0.9rem',
        fontWeight: 'bold',
    },
})(Typography);

const Regions = ({
    selectedValue,
    setValue,
    ...props
}) => {

    return (
        <div>
            <StyledTypography id="non-linear-slider" gutterBottom>
                Select region
            </StyledTypography>
            <Select
                {...props}
                data-testid="drop-down"
                displayEmpty
                value={selectedValue}
                onChange={setValue}
            >
                <MenuItem key="0" value={""}>Select region</MenuItem>
                {
                    options.regions.map((region, regionIndex) => {
                        return <MenuItem key={region} value={regionIndex}>{region}</MenuItem>
                    })
                }
            </Select>
        </div>
    );
}

Regions.propTypes = {
    selectedValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    setValue: PropTypes.func,
}

Regions.defaultProps = {
    selectedValue: "",
    setValue: Function.prototype,
}

const StyledRegions = withStyles({
    select: {
        width: 300,
        margin: 10,
    },
})(Regions);

export default StyledRegions;
