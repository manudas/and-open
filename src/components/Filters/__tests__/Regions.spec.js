import {
    render,
    fireEvent,
    waitFor
} from '@testing-library/react';
import { act } from "react-dom/test-utils";

import Regions from '../Regions';

describe('Test suite for Regions', () => {

    beforeEach(() => {
        jest.useFakeTimers();
    });

    it('should render and match snapshot', () => {
        let renderer;
        act(() => {
            renderer = render(<Regions />);
            jest.runAllTimers();
        });
        const { container } = renderer;
        expect(container).toMatchSnapshot();
    });

    it('should unfold when clicked and call setValue function if selected value changes', async () => {
        const setRegion = jest.fn();
        let renderer;
        act(() => {
            renderer = render(<Regions setValue={setRegion} />);
            jest.runAllTimers();
        });

        const {
            queryByTestId,
            getByRole
        } = renderer;

        const dropDown = queryByTestId('drop-down');

        fireEvent.mouseDown(dropDown.firstChild);

        let listBox;
        await waitFor(() => listBox = getByRole('listbox'));

        expect(listBox).toBeInTheDocument();

        const dropDownElement = listBox.children[1];

        // Simulate clicking on the United States filter
        fireEvent.click(dropDownElement);

        expect(setRegion).toHaveBeenCalledTimes(1)
    });
});
