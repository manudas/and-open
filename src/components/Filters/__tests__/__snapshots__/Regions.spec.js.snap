// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Regions should render and match snapshot 1`] = `
<div>
  <div>
    <p
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-2 MuiTypography-body1 MuiTypography-gutterBottom"
      id="non-linear-slider"
    >
      Select region
    </p>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline"
      data-testid="drop-down"
    >
      <div
        aria-haspopup="listbox"
        class="MuiSelect-root MuiSelect-select Regions-select-1 MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        role="button"
        tabindex="0"
      >
        Select region
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
